src/app/pages/authentication/registration/registration.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, ValidationErrors, ValidatorFn, Validators } from '@angular/forms';
+import { ToastrService } from 'ngx-toastr';
+import { UserService } from '../../../core/services/user.service';
+import { Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.scss'],
+})
+export class RegistrationComponent implements OnInit, OnDestroy {
+  registrationForm: FormGroup;
+
+  subscription: Subscription;
+
+  mouseoverLogin: boolean;
+
+  private userName: FormControl;
+
+  private userPassword: FormControl;
+
+  private confirmPassword: FormControl;
+
+  private userEmail: FormControl;
+
+  constructor(private readonly fb: FormBuilder, private readonly toastr: ToastrService, private readonly userService: UserService, private readonly router: Router) {
+  }
+
+  ngOnInit(): void {
+    this.userName = new FormControl('', [Validators.required]);
+    this.userPassword = new FormControl('', [Validators.required, Validators.minLength(6)]);
+    this.confirmPassword = new FormControl('', [Validators.required, Validators.minLength(6)]);
+    this.userEmail = new FormControl('', [Validators.required]);
+
+    this.registrationForm = this.fb.group({
+      userName: this.userName,
+      userEmail: this.userEmail,
+      passwords: this.fb.group({
+        userPassword: this.userPassword,
+        confirmPassword: this.confirmPassword,
+      }, { validators: this.checkPasswords }),
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.subscription?.unsubscribe();
+  }
+
+  checkPasswords: ValidatorFn = (): ValidationErrors | null => {
+    let pass = this.userPassword.value;
+    let confirmPass = this.confirmPassword.value;
+    return pass === confirmPass ? null : { notSame: true };
+  };
+
+
+  handleValidateEmail() {
+    return (this.userEmail.touched || this.mouseoverLogin) && this.userEmail.invalid;
+  }
+
+  handleCancelButtonClick() {
+    this.router.navigate(['']);
+  }
+
+  handleValidatePassword() {
+    return (this.userPassword.touched || this.mouseoverLogin) && this.userPassword.errors?.['required'];
+  }
+
+  handleValidateConfirmPassword() {
+    return (this.confirmPassword.touched || this.mouseoverLogin) && this.confirmPassword.errors?.['required'];
+  }
+
+  handleValidateLogin() {
+    return (this.userName.touched || this.mouseoverLogin) && this.userName.invalid;
+  }
+
+  handleRegisterUser() {
+    const formValue = this.registrationForm.value;
+    const data = {
+      id: 6,
+      userName: formValue.userName,
+      userPassword: this.registrationForm.get('passwords')?.get('userPassword')?.value,
+      userEmail: formValue.userEmail,
+    };
+
+    this.subscription = this.userService.addUser(data).subscribe(users => users);
```

Comment:
Do you need to get users in this component?
Here is the same situation as with login. You should move out this logic to the service
```
this.userService.registerUser(data).subscibe(()=> {
//handle success
},
()=> {
//handle error
})
```